keyring_darwin.go

```diff
@@ -45,18 +48,26 @@ func (k macOSXKeychain) Get(service, username string) (string, error) {
 		}
 		return "", err
 	}
-	// if the added secret has multiple lines, osx will hex encode it
+
 	trimStr := strings.TrimSpace(string(out[:]))
-	dec, err := hex.DecodeString(trimStr)
-	// if there was an error hex decoding the string, assume it's not encoded
-	if err != nil {
-		return fmt.Sprintf("%s", trimStr), nil
+	// if the string has the well-known prefix, assume it's encoded
+	if strings.HasPrefix(trimStr, encodingPrefix) {
+		if dec, err := hex.DecodeString(trimStr[len(encodingPrefix):]); err == nil {
```

Comment:
if `err != nil` then we should return an error.

Reply:
Something like https://github.com/zalando/go-keyring/pull/37/commits/c90858f20438ddd36bc9cacb5a9df6eba2ab30e5?